### ISSUE TITLE
JobSrv gets in infinite restart loop

### ISSUE THREAD

**Issue by chefsalim** [habitat-sh/habitat#3627]
It get's into a bad state and the process keeps restarting - not clear why this is happening.

```
Oct 08 21:14:43 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
Oct 08 21:14:44 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 08 21:14:49 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 08 21:14:49 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 08 21:14:49 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 08 21:14:49 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
Oct 08 21:14:50 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 08 21:14:55 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 08 21:14:55 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 08 21:14:55 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 08 21:14:55 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
Oct 08 21:14:56 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 08 21:15:01 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 08 21:15:01 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 08 21:15:01 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 08 21:15:01 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
Oct 08 21:15:02 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 08 21:15:07 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 08 21:15:07 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 08 21:15:07 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 08 21:15:07 ip-10-0-0-172 hab[6640]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
```

**Comment by chefsalim** [comment]
This is still reproing:
```
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: hab-sup(MR): Starting core/builder-jobsrv
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: hab-sup(MR): Starting gossip-listener on 0.0.0.0:9638
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: hab-sup(MR): Starting http-gateway on 0.0.0.0:9631
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(HK): Hooks compiled
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(SR): Initializing
Oct 11 13:51:38 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 11 13:51:39 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:39 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:39 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(HK): Hooks compiled
Oct 11 13:51:40 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:40 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:41 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:41 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:42 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:42 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:43 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:43 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(SV): Starting service as user=hab, group=hab
Oct 11 13:51:44 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:45 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:45 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(HK): Hooks compiled
Oct 11 13:51:45 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:46 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:46 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:47 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:47 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:48 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:48 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(HK): Hooks compiled
Oct 11 13:51:48 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:49 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:49 ip-10-0-0-172 hab[5721]: Could not send rumor to "4194275183934b3695e6a0ed283961ec" @ "10.0.0.109:9638"; ZMQ said: Resource temporarily unavailable
Oct 11 13:51:50 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for log data on tcp://0.0.0.0:5568
Oct 11 13:51:50 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for commands on tcp://0.0.0.0:5566
Oct 11 13:51:50 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Listening for heartbeats on tcp://0.0.0.0:5567
Oct 11 13:51:50 ip-10-0-0-172 hab[5721]: builder-jobsrv.acceptance(O): Failed to route request, no reachable RouteSrv```

**Comment by reset** [comment]
Fixed by #3738
